### PR TITLE
Clarify that UKI only has to have .linux section

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: specs/unified_kernel_image.md [hunk 1]
```diff
@@ -32,9 +32,11 @@ The file format is PE/COFF (Portable Executable / Common Object File Format). Th
...
 and UKIs follow the standard, so exact details will not be repeated here.
 
-UKIs can be generated via a [single, relatively simple `objcopy`
-invocation](https://www.freedesktop.org/software/systemd/man/systemd-stub.html#Assembling%20Kernel%20Images),
-that glues the listed components together, generating one PE binary that then can be signed for SecureBoot.
+UKIs are created as a PE/COFF file with additional sections that glues together all the components.
+In principle this can be done with a relatively simple `objcopy` invocation,
+but the recommended way is to use a helper program
+([`ukify`](https://www.freedesktop.org/software/systemd/man/ukify.html)),
+which takes care of appropriate alignment and facilitates signing for SecureBoot.
```
Comment (poettering) [inline specs/unified_kernel_image.md:39]: my PR #81 also adds this btw.

**Comment by poettering** [comment]
please rebase

File: specs/unified_kernel_image.md [hunk 2]
```diff
@@ -55,7 +55,16 @@ UKIs consist of the following resources:
...
+Note that all of the sections defined above are singletons:
+they may appear at most once,
+except for the `.dtb` section which may be appear many times.
+
+Only the `.linux` section is required for the image to be considered a Unified *Kernel* Image.
+A UKI will generally also contain various sections required for the boot stub,
+but we don't document those here.
+Note that the same file format is also used for other purposes,
+for example addons,
+which will contain a different subset of sections.
```
Comment (poettering) [inline specs/unified_kernel_image.md:67]: this is very very weird line breaking

File: specs/unified_kernel_image.md [hunk 3]
```diff
@@ -55,7 +55,16 @@ UKIs consist of the following resources:
 * Optionally, a JSON file encoding expected PCR 11 hash values seen from userspace once the UKI has booted up, along with signatures of these expected PCR 11 hash values, in the `.pcrsig` section. The signatures must also match the abovementioned key pair.
 * Optionally, a CSV file encoding the SBAT metadata for the image, in the `.sbat` section. The [SBAT format is defined by the Shim project](https://github.com/rhboot/shim/blob/main/SBAT.md), and used for UEFI revocation purposes.
 
-Note that all of the sections defined above are singletons: they may appear once at most – except for the `.dtb` section which may be appear more than once.
+Note that all of the sections defined above are singletons:
+they may appear at most once,
+except for the `.dtb` section which may be appear many times.
```
Comment (poettering) [inline specs/unified_kernel_image.md:60]: many? multiple!

File: specs/unified_kernel_image.md [hunk 4]
```diff
@@ -55,7 +55,16 @@ UKIs consist of the following resources:
...
+
+Only the `.linux` section is required for the image to be considered a Unified *Kernel* Image.
```
Comment (poettering) [inline specs/unified_kernel_image.md:62]: maybe say explicitly here that boot menus such as sd-boot might have stricter requirements, i.e. require .osrel or so. But be vague here, I think that's something we should document in sd-boot documents, not here in these docs

**Comment by poettering** [comment]
Maybe close #74 with this?

**Comment by keszybz** [comment]
Rebased and updated as requested.

File: specs/unified_kernel_image.md [hunk 5]
```diff
@@ -31,9 +31,11 @@ The file format is PE/COFF (Portable Executable / Common Object File Format). Th
 industry-standard file format, used for example in UEFI environments,
 and UKIs follow the standard, so exact details will not be repeated here.
 
-UKIs can be generated via a [single, relatively simple `objcopy`
-invocation](https://www.freedesktop.org/software/systemd/man/systemd-stub.html#Assembling%20Kernel%20Images),
-that glues the listed components together, generating one PE binary that then can be signed for SecureBoot.
+UKIs are created as a PE/COFF file with additional sections that glues together all the components.
```
Comment (DaanDeMeyer) [inline specs/unified_kernel_image.md:34]: "with additional sections that glues together all the components." => I don't really get what this is supposed to say, maybe try to reword?

File: specs/unified_kernel_image.md [hunk 6]
```diff
@@ -53,9 +55,19 @@ UKIs consist of the following resources:
 * Optionally, a JSON file encoding expected PCR 11 hash values seen from userspace once the UKI has booted up, along with signatures of these expected PCR 11 hash values, in the `.pcrsig` section. The signatures must also match the key pair described below.
 * Optionally, the public part of a public-private key pair in PEM format used to sign the expected PCR 11 value of the image, in the `.pcrpkey` section.
 
-Note that all of the sections defined above are singletons: they may
-appear once at most – except for the `.dtb` section which may
-appear more than once.
+Note that all of the sections defined above are singletons:
+they may appear at most once,
+except for the `.dtb` section which may be appear multiple times.
```
Comment (DaanDeMeyer) [inline specs/unified_kernel_image.md:60]: "may be appear"?

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: specs/unified_kernel_image.md (hunk 5 above)
Comment (keszybz) [reply specs/unified_kernel_image.md]: Reworded. It's not so easy to keep this short but not introduce inaccuracies.

**Pull request merged**